frontend/src/metabase-lib/compat/compat.ts

```diff
@@ -0,0 +1,96 @@
+import * as ML from "metabase-lib";
+import type { FieldFilter } from "metabase-types/api";
+import type StructuredQuery from "metabase-lib/queries/StructuredQuery";
+import LegacyFilter from "metabase-lib/queries/structured/Filter";
+
+export const getOperatorsMap = ({
+  query,
+  column,
+  stageIndex = -1,
+}: {
+  query?: ML.Query;
+  column?: ML.ColumnWithOperators;
+  stageIndex?: number;
+}) => {
+  if (!query || !column) {
+    return {};
+  }
+
+  const operators = ML.filterableColumnOperators(column);
+  return Object.fromEntries(
+    operators.map(operator => [
+      ML.displayInfo(query, stageIndex, operator).shortName,
+      operator,
+    ]),
+  );
+};
+
+export const getMlv2FilterClause = (filter: LegacyFilter, stageIndex = -1) => {
+  const mlv2Query = filter.query().question()._getMLv2Query();
+
+  const appliedFilterClause = ML.findFilterForLegacyFilter(
+    mlv2Query,
+    stageIndex,
+    [...(filter as unknown as FieldFilter)],
+  );
```

Comment:
We should be able to replace `[...(filter as unknown as FieldFilter)]` with something like that:

```ts
class Filter extends MBQLClause {
  raw(): FieldFilter {
    return [...this]
  }
}

ML.findFilterForLegacyFilter(
    mlv2Query,
    stageIndex,
    filter.raw(),
  );
```